### PR TITLE
Added Basic Primitives, fixed With() bug.

### PR DESCRIPTION
**Pull request by jalaja-git**
@rchamarthy

### CODE REVIEW

File: log_test.go [hunk 1]
```diff
@@ -13,5 +14,25 @@ func TestLogger(t *testing.T) {
 		So(l.LogLevel(), ShouldEqual, Debug)
 		So(l.Test(Debug), ShouldBeTrue)
 		l.Info().Str("hello", "world").Msg("Test")
+		l.Info().Int("Int value", 1000).Msg("Printing Int")
+		l.Info().Msg("Printing Message only")
+		l.Info().Err(errors.New("Error event")).Msg("Printing Error event")
+
+		log := l.With().Str("with_key", "with_val").Logger()
+		log.Info().Msg("Str - With Testing")
```
Comment (rchamarthy) [inline log_test.go]: Lets provide a buffer writer and assert on the output. Otherwise we dont get coverage

File: zerolog.go [hunk 2]
```diff
@@ -103,11 +110,30 @@ func (c ctxImpl) Logger() Logger {
 }
 
 func (c ctxImpl) Str(key string, val string) Context {
-	c.Context.Str(key, val)
+	c.Context = c.Context.Str(key, val)
+	c.l.Logger = c.Context.Logger()
```
Comment (rchamarthy) [inline zerolog.go]: I think we should do that when our Context.Logger() is called. Else we will keep copying for every chained call. For example: log.With().Str("a", "b").Str("c", "d").Logger() will call zerolog.Logger() three times and each time the context is re-copied.

File: log_test.go (hunk 1 above)
Comment (jalaja-git) [reply log_test.go]: Need to figure how to do this. This is the current coverage..
go test -coverprofile cover.out
...
coverage: 84.6% of statements
ok      github.com/rs/zlog      0.012s
JGANAPAT-M-H0MN:zlog jganapat$

File: zerolog.go (hunk 2 above)
Comment (jalaja-git) [reply zerolog.go]: Nice, fixed this. 
..
func (c ctxImpl) Logger() Logger {
	c.l.Logger = c.Context.Logger()
	return c.l
}

**Pull request merged**